File: algorithm/golang/contains_duplicate.go
package algo

import "sort"

func ContainsDuplicate(nums []int) bool {
	set := make(map[int]bool)
	for _, num := range nums {
		set[num] = true
	}
	return len(nums) != len(set)
}

func ContainsDuplicate2(nums []int) bool {
	sort.Ints(nums)
	for i := 1; i < len(nums); i++ {
		if nums[i] == nums[i-1] {
			return true
		}
	}
	return false
}

func ContainsDuplicate3(nums []int) bool {
	m := make(map[int]bool)
	for _, v := range nums {
		if m[v] {
			return true
		}
		m[v] = true
	}
	return false
}

File: algorithm/typescript/src/contains_duplicate.ts
export function contains_duplicate(nums: number[]): boolean {
    return new Set<number>(nums).size !== nums.length;
}

export function contains_duplicate2(nums: number[]): boolean {
    nums = nums.sort()
    for (let i = 0; i < nums.length - 1; i++) {
        if (nums[i] === nums[i + 1]) {
            return true;
        }
    }
    return false;
}

export function contains_duplicate3(nums: number[]): boolean {
    const set = new Set<number>();
    for (const num of nums) {
        if (set.has(num)) {
            return true;
        }
        set.add(num);
    }
    return false;
}

